src/app/cdk/side-bar/modals/modal-keyword/modal-keyword.component.ts

```diff
@@ -0,0 +1,185 @@
+import { RecordKeywordService } from 'src/app/core/record-keyword/record-keyword.service'
+import { UserRecord } from 'src/app/types/record.local'
+import { VisibilityStrings } from 'src/app/types/common.endpoint'
+import { KeywordEndPoint } from 'src/app/types/record-keyword.endpoint'
+
+@Component({
+  selector: 'app-modal-keyword',
+  templateUrl: './modal-keyword.component.html',
+  styleUrls: [
+    './modal-keyword.component.scss-theme.scss',
+    './modal-keyword.component.scss',
+  ],
+})
+export class ModalKeywordComponent implements OnInit, OnDestroy {
+  $destroy: Subject<boolean> = new Subject<boolean>()
+
+  addedKeywordsCount = 0
+  userRecord: UserRecord
+  keywordsForm: FormGroup
+  keywords: Assertion[]
+  defaultVisibility: VisibilityStrings
+  originalBackendKeywords: KeywordEndPoint
+  isMobile: boolean
+  loadingKeywords = true
+  userSession: UserSession
+  platform: PlatformInfo
+
+  ngOrcidKeyword = $localize`:@@topBar.keyword:Keyword`
+
+  constructor(
+    @Inject(WINDOW) private window: Window,
+    @Inject(MAT_DIALOG_DATA) public data: UserRecord,
+    public dialogRef: MatDialogRef<ModalComponent>,
+    private _recordKeywordService: RecordKeywordService,
+    private _changeDetectorRef: ChangeDetectorRef,
+    private _platform: PlatformInfoService,
+    private _userService: UserService
+  ) {
+    this._platform
+      .get()
+      .pipe(takeUntil(this.$destroy))
+      .subscribe((platform) => {
+        this.platform = platform
+        this.isMobile = platform.columns4 || platform.columns8
+      })
+    this._userService
+      .getUserSession()
+      .pipe(takeUntil(this.$destroy))
+      .subscribe((userSession) => {
+        this.userSession = userSession
+      })
+  }
+
+  ngOnInit(): void {
+    this._recordKeywordService
+      .getKeywords()
+      .pipe(first())
+      .subscribe((keywords: KeywordEndPoint) => {
+        this.defaultVisibility = keywords.visibility.visibility
+        this.originalBackendKeywords = cloneDeep(keywords)
+        this.keywords = this.originalBackendKeywords.keywords
+        this.backendJsonToForm(this.originalBackendKeywords)
+        this.loadingKeywords = false
+      })
+  }
+
+  backendJsonToForm(keywordEndpointJson: KeywordEndPoint) {
+    const keywords = keywordEndpointJson.keywords
+    const group: { [key: string]: FormGroup } = {}
+
+    keywords.forEach((keyword) => {
+      group[keyword.putCode] = new FormGroup({
+        content: new FormControl(keyword.content),
+        visibility: new FormControl(keyword.visibility.visibility, {}),
+      })
+    })
+    this.keywordsForm = new FormGroup(group)
+  }
+
+  formToBackend(keywordsForm: FormGroup): KeywordEndPoint {
+    const keywords = {
+      errors: [],
+      keywords: [],
+      visibility: this.originalBackendKeywords.visibility,
+    }
+    const toBackendKeywords = cloneDeep(this.keywords)
+    toBackendKeywords.reverse()
```

Comment:
we need to call `_changeDetectorRef.detach()` before the reverse. 

This will avoid showing the list reversed on the UI while is been send to the backend. 

I have added this change to all the other modals on my last PR.